Sources/Event.swift

```diff
@@ -816,26 +816,18 @@ extension Signal.Event {
 				}
 
 				schedulerDisposable.inner = scheduler.schedule(after: scheduleDate) {
-					let pendingValue: Value? = state.modify { state in
-						defer {
-							if state.pendingValue != nil {
-								state.pendingValue = nil
-								state.previousDate = scheduleDate
-							}
-						}
-						return state.pendingValue
-					}
-
-					if let pendingValue = pendingValue {
+					if let pendingValue = state.modify({ $0.retrieveValue(date: scheduleDate) }) {
 						action(.value(pendingValue))
 					}
 				}
 			}
 		}
 	}
 
-	internal static func debounce(_ interval: TimeInterval, on scheduler: DateScheduler) -> Transformation<Value, Error> {
+	internal static func debounce(_ interval: TimeInterval, on scheduler: DateScheduler, discardsWhenCompleted: Bool) -> Transformation<Value, Error> {
```

Comment:
I think this should probably be `discardWhenCompleted` (no _s_).

Reply:
I feel like that gives the implication that it will always discard *something*. Might be worth bikeshedding.

Reply:
I don't see how the _s_ changes that.

The _s_ seems inconsistent the base operator name and other operators. Since it's `debounce`, not `debounces`, I think `discardWithCompleted` is more appropriate.